internal/services/workloads/workloads_sap_virtual_instance.go

```diff
@@ -0,0 +1,1001 @@
+		MaxItems: 1,
+		Elem: &pluginsdk.Resource{
+			Schema: map[string]*pluginsdk.Schema{
+				"image_reference": {
+					Type:     pluginsdk.TypeList,
+					Required: true,
+					ForceNew: true,
+					MaxItems: 1,
+					Elem: &pluginsdk.Resource{
+						Schema: map[string]*pluginsdk.Schema{
+							"offer": {
+								Type:         pluginsdk.TypeString,
+								Required:     true,
+								ForceNew:     true,
+								ValidateFunc: validation.StringIsNotEmpty,
+							},
+
+							"publisher": {
+								Type:     pluginsdk.TypeString,
+								Required: true,
+								ForceNew: true,
+								ValidateFunc: validation.StringInSlice([]string{
+									"RedHat",
+									"SUSE",
+								}, false),
+							},
+
+							"sku": {
+								Type:         pluginsdk.TypeString,
+								Required:     true,
+								ForceNew:     true,
+								ValidateFunc: validation.StringIsNotEmpty,
+							},
+
+							"version": {
+								Type:         pluginsdk.TypeString,
+								Required:     true,
+								ForceNew:     true,
+								ValidateFunc: validation.StringIsNotEmpty,
+							},
+						},
+					},
+				},
+
+				"os_profile": {
+					Type:     pluginsdk.TypeList,
+					Required: true,
+					ForceNew: true,
+					MaxItems: 1,
+					Elem: &pluginsdk.Resource{
+						Schema: map[string]*pluginsdk.Schema{
+							"admin_username": {
+								Type:         pluginsdk.TypeString,
+								Required:     true,
+								ForceNew:     true,
+								ValidateFunc: validate.AdminUsername,
+							},
+
+							"ssh_key_pair": {
+								Type:     pluginsdk.TypeList,
+								Required: true,
+								ForceNew: true,
+								MaxItems: 1,
+								Elem: &pluginsdk.Resource{
+									Schema: map[string]*pluginsdk.Schema{
+										"private_key": {
+											Type:         pluginsdk.TypeString,
+											Required:     true,
+											ForceNew:     true,
+											Sensitive:    true,
+											ValidateFunc: validation.StringIsNotEmpty,
+										},
+
+										"public_key": {
+											Type:         pluginsdk.TypeString,
+											Required:     true,
+											ForceNew:     true,
+											ValidateFunc: validation.StringIsNotEmpty,
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+
+				"vm_size": {
+					Type:         pluginsdk.TypeString,
+					Required:     true,
+					ForceNew:     true,
+					ValidateFunc: validation.StringIsNotEmpty,
+				},
+			},
+		},
+	}
+}
+
+func SchemaForSAPVirtualInstanceDiskVolumeConfiguration() *pluginsdk.Schema {
+	return &pluginsdk.Schema{
+		Type:     pluginsdk.TypeSet,
+		Optional: true,
+		ForceNew: true,
+		Elem: &pluginsdk.Resource{
+			Schema: map[string]*pluginsdk.Schema{
+				"volume_name": {
+					Type:     pluginsdk.TypeString,
+					Required: true,
+					ForceNew: true,
+					ValidateFunc: validation.StringInSlice([]string{
+						"backup",
+						"hana/data",
+						"hana/log",
+						"hana/shared",
+						"os",
+						"usr/sap",
+					}, false),
+				},
+
+				"count": {
```

Comment:
What is this the `count` of?

Reply:
@tombuildsstuff , I updated to `number_of_disks`.

Reasons - why was this issue marked as resolved?:
1. There is no response for a long time from the reviewer. 
2. The next reviewer also doesn't mention that my point is incorrect.
3. Sometimes seems the reviewers also don't resolve the conversions even if the reviewer agrees my point in previous PRs.

Based on above reasons, so I resolve the conversion recently. Note: The conversion isn't resolved at the beginning.

cc @mybayern1974

Reply:
Updated to number_of_disks